crates/runtime/src/dataconnector/listing/connector.rs

```diff
@@ -139,50 +142,74 @@ pub trait ListingTableConnector: DataConnector {
+            (Some("jsonl"), _) | (None, Some("jsonl"))=> Ok((
+                Some(Arc::new(JsonFormat::default())),
+                extension.unwrap_or(".jsonl".to_string()),
+            )),
+            (Some("parquet"), _) | (None, Some("parquet"))=> Ok((
                 Some(Arc::new(
                     ParquetFormat::default().with_options(self.get_table_parquet_options(dataset)?),
                 )),
                 extension.unwrap_or(".parquet".to_string()),
             )),
-            Some(format) => Ok((None, format!(".{format}"))),
-            None => {
-                if let Some(ext) = std::path::Path::new(dataset.path().as_str()).extension() {
-                    if ext.eq_ignore_ascii_case("csv") {
-                        return Ok((
-                            Some(self.get_csv_format(dataset, params)?),
-                            extension.unwrap_or(".csv".to_string()),
-                        ));
-                    }
-                    if ext.eq_ignore_ascii_case("parquet") {
-                        return Ok((
-                            Some(Arc::new(
-                                ParquetFormat::default()
-                                    .with_options(self.get_table_parquet_options(dataset)?),
-                            )),
-                            extension.unwrap_or(".parquet".to_string()),
-                        ));
-                    }
-                }
-
-                Err(
+            (Some(format), _) => Ok((None, format!(".{format}"))),
+            (_, _) => Err(
                     crate::dataconnector::DataConnectorError::InvalidConfiguration {
                         dataconnector: format!("{self}"),
                         message: "The required 'file_format' parameter is missing.\nEnsure the parameter is provided, and try again.".to_string(),
                         connector_component: ConnectorComponent::from(dataset),
                         source: "Missing file format".into(),
                     },
-                )
-            }
+                ),
         }
     }
 
+    /// Returns a [`JsonFormat`] based on the provided [`Datasets`] parameters.
+    ///
+    /// If the [`Dataset`] has the relevant parameter, return an error if the value is invalid.
+    fn get_jsonl_format(
+        &self,
+        dataset: &Dataset,
+        params: &Parameters,
+    ) -> DataConnectorResult<Arc<JsonFormat>>
+    where
+        Self: Display,
+    {
+        let mut format = JsonFormat::default();
+
+        if let ParamLookup::Present(comp_as_secret) = params.get("compression") {
+            let compression = comp_as_secret.expose_secret().parse::<FileCompressionType>().boxed().context(crate::dataconnector::InvalidConfigurationSnafu {
+                    dataconnector: format!("{self}"),
+                    message: format!(
+                        "Invalid JSONL compression_type: {comp_as_secret:?}, supported types are: GZIP, BZIP2, XZ, ZSTD, UNCOMPRESSED"),
```

Comment:
We should in general try to stick to the Display representations of objects in error messages